### PR TITLE
Us9 1 create swap

### PR DESCRIPTION
**Pull request by CECDO**
creation of the swap form

### CODE REVIEW

**Review by Bruno-Fernandes-37 — CHANGES_REQUESTED** [review]
overriding methods that should not be in swapController, no version file ..... no big deal for sure

**Review by ghost — DISMISSED** [review]

**Review by Bruno-Fernandes-37 — DISMISSED** [review]

**Review by RomainClair — CHANGES_REQUESTED** [review]
The fixtures need to be fixed according to the database changes

**Comment by Benjamin-51** [comment]
> The fixtures need to be fixed according to the database changes

What fixtures would we need to change? We are unsure about if this current PR requires changes or if we need to update them like adding skills and comments to our users.

**Review by Bruno-Fernandes-37 — DISMISSED** [review]
it all looks good for me

**Review by Bruno-Fernandes-37 — APPROVED** [review]

File: src/Controller/SwapController.php [hunk 1]
```diff
@@ -34,4 +42,44 @@ public function display(Skill $skill, User $user): Response
...
+        $swap = new Swap();
+        $form = $this->createForm(SwapType::class, $swap);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $currentUser = $this->getUser();
+            if ($currentUser instanceof User) {
+                $swap->setAsker($currentUser);
+                $swap->setHelper($helper);
+                $swap->setSkill($skill);
+                $swap->setDate(new DateTime());
```
Comment (RomainClair) [inline src/Controller/SwapController.php:66]: This could have been done in the Swap construct

File: src/Entity/Swap.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private \DateTimeInterface $date;
+
+    /**
+     * @ORM\Column(type="text")
+     */
+    private string $message;
```
Comment (RomainClair) [inline src/Entity/Swap.php:49]: You probably should have constraints to validate the message... May be avoiding empty messages ?

**Review by RomainClair — APPROVED** [review]

**Pull request merged**